### ISSUE TITLE
Rootless pull fails with - failed to register layer: failed to Lchown <file>: invalid argument

### ISSUE THREAD

**Issue by HendrikSchrieber** [moby/moby#45757]
### Description

Trying to pull the image ghcr.io/m9sweeper/dash:1.3.5 fails with the given error message. Increasing subuid and subgid to 1000000000 in /etc/subuid and /etc/subgid, as the error message suggests, does not solve the issue.

Docker is running in rootless mode set-up according to official documentation: https://docs.docker.com/engine/security/rootless/

This is the first image where the issue appeared, I have pulled many images where pulling works fine.

The image is from this project: https://github.com/m9sweeper/m9sweeper.

### Reproduce

Use docker rootless daemon to pull the given image:

```
$ docker pull ghcr.io/m9sweeper/dash:1.3.5
1.3.5: Pulling from m9sweeper/dash
f56be85fc22e: Already exists 
4e21e25411da: Pull complete 
9890dc0be345: Pull complete 
61d1890b04b0: Pull complete 
f45381a0d065: Pull complete 
640e7c28ec7b: Pull complete 
55d04ba527cb: Extracting  66.16MB/66.16MB
82f81d286bff: Download complete 
12c0946e17d7: Download complete 
8a406b2e38b4: Download complete 
b2f1887489a8: Download complete 
failed to register layer: failed to Lchown "/usr/src/app/node_modules/buffer-equal-constant-time/.npmignore" for UID 718322462, GID 454177323 (try increasing the number of subordinate IDs in /etc/subuid and /etc/subgid): lchown /usr/src/app/node_modules/buffer-equal-constant-time/.npmignore: invalid argument
```

### Expected behavior

Successful pull of the image.

### docker version

```bash
Client: Docker Engine - Community
 Version:           24.0.2
 API version:       1.43
 Go version:        go1.20.4
 Git commit:        cb74dfc
 Built:             Thu May 25 21:51:00 2023
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          24.0.2
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.4
  Git commit:       659604f
  Built:            Thu May 25 21:51:00 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.21
  GitCommit:        3dce8eb055cbb6872793272b4f20ed16117344f8
 runc:
  Version:          1.1.7
  GitCommit:        v1.1.7-0-g860f061
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
 rootlesskit:
  Version:          1.1.0
  ApiVersion:       1.1.1
  NetworkDriver:    slirp4netns
  PortDriver:       builtin
  StateDir:         /tmp/rootlesskit1864896339
 slirp4netns:
  Version:          1.0.1
  GitCommit:        6a7b16babc95b6a3056b33fb45b74a6f62262dd4
```

### docker info

```bash
Client: Docker Engine - Community
 Version:    24.0.2
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.5
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.18.1
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 49
  Running: 0
  Paused: 0
  Stopped: 49
 Images: 335
 Server Version: 24.0.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: false
  userxattr: true
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  rootless
  cgroupns
 Kernel Version: 5.19.0-43-generic
 Operating System: Ubuntu 22.04.2 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.33GiB
 Name: DEV-BHV-UL003
 ID: HVOU:MRXS:3XZL:UC5F:4FVD:GYUH:UQEN:24IB:3UE6:4PIJ:WONU:HWBF
 Docker Root Dir: /home/hendrik/.local/share/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Comment by jacobbeasley** [comment]
We put a fix in m9sweeper for this - looks like one of the npm packages was using a strange UID

https://github.com/m9sweeper/m9sweeper/pull/134/files

**Comment by thaJeztah** [comment]
I wonder where those UID's came from; they almost feel like somehow files were copied from a container that was running with user-namespaces enabled (remapped IDs), and those IDs were not converted back.

The fix in https://github.com/m9sweeper/m9sweeper/pull/134 may be tapering over a problem.

**Comment by jacobbeasley** [comment]
One of the packages installed with npm seems to download a file owned by a user with a high user id. We implemented a fix for this by just change our dockerfile to "chown" it with the proper user id to avoid this issue.

**Comment by jacobbeasley** [comment]
So it would seem one of the users was "too high" :D

**Comment by thaJeztah** [comment]
Thanks! Haha, yes.

Would be interesting to learn what happened with that package (if you're able to narrow down which one has the issue). I guess the workaround is not too bad, as it's chown'ing in the same layer as the files are added (so not duplicating the layer's size), but it's possible that other users of that package may be doing it in a separate `RUN` (and now duplicating those files without realising, and thinking "why is my image so large?").

**Comment by jacobbeasley** [comment]
The npm package name is in the filename: buffer-equal-constant-time/

**Comment by jacobbeasley** [comment]
We are in the process of removing strange/obscure packages

**Comment by AkechiShiro** [comment]
@jacobbeasley has there been any news so far ?

**Comment by kyle-intelletive** [comment]
@AkechiShiro The issue with the m9sweeper issue should be resolved by using a newer version. The current latest tag is 1.6.0, and the issue was fixed in 1.4.0
